source/scale/redis.rst

```diff
@@ -0,0 +1,115 @@
+        "CacheType": "redis",
+        "RedisAddress": "redis.example.com:6379",
+        "RedisPassword": "",
+        "RedisCachePrefix": "",
+        "RedisDB": -1,
+        "DisableClientCache": false
+    }
+
+Deploy with AWS ElastiCache
+---------------------------
+
+For enterprise-scale deployments, AWS ElastiCache for Redis provides a fully managed Redis service that simplifies deployment, operation, and scaling. 
+
+When deploying Mattermost with AWS ElastiCache:
+
+1. Create an ElastiCache for Redis instance, choosing Redis OSS version 7.1 or later.
+2. Based on load testing results, we recommend starting with a ``cache.m7g.2xlarge`` instance for deployments supporting over 100,000 users.
+3. Configure your Mattermost server to connect to the ElastiCache endpoint.
+4. Monitor CPU utilization closely, as Redis is single-threaded and performance bottlenecks are typically CPU-bound rather than memory-bound.
+
+Performance considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using Redis with Mattermost, consider the following performance insights:
+
+- Current load tests indicate that memory usage of Redis is relatively low, with CPU being the primary bottleneck because Redis is single-threaded.
+- When planning your Redis deployment, use CPU utilization as the primary metric for scaling decisions.
+- For extremely large deployments, consider sharding your Redis deployment to distribute the workload.
+
+Implementation details
+----------------------
+
+Mattermost uses the `rueidis <https://github.com/redis/rueidis>`_ library instead of the more popular `go-redis <https://github.com/redis/go-redis>`_, primarily because rueidis offers more performance optimizations out of the box, including:
+
+- Automatic pipelining of concurrent queries
+- Client-side caching
+- Additional performance tuning options like MaxFlushDelay
+
+The primary implementation files include:
+
+- ``platform/services/cache/provider.go`` - Contains the library initialization code
+- ``platform/services/cache/redis.go`` - Contains the cache interface implementation
+
+Frequently asked questions (FAQ)
+--------------------------------
+
+Do I need to use Redis?
+~~~~~~~~~~~~~~~~~~~~~~~
+
+While Redis is not required for smaller Mattermost deployments, it is highly recommended for large-scale installations.
+- If your installation has more than 100K users, Redis is highly recommended. Redis significantly improves performance by reducing database load and optimizing common operations.
+- If your installation has less than 100K users, Redis is not recommended.
+
+It is suggested to keep an eye on the `sum(rate(mattermost_cache_mem_invalidation_total[5m]))` metric which indicates the rate of cache invalidations happening. A single cache invalidation will send a message across your HA cluster to all nodes, leading to database calls from all of them. So the metric, combined with the number of nodes in your cluster should be used to determine when is the right time to use Redis.
```

Comment:
Thoughts on making `sum(rate(mattermost_cache_mem_invalidation_total[5m]))` code formatted rather than italic?

![image](https://github.com/user-attachments/assets/aeb7abd7-53a7-4a41-b0cb-3c0b4f68d53d)

Reply:
Done